gax/src/main/java/com/google/api/gax/rpc/TransportChannelProvider.java

```diff
@@ -64,6 +64,7 @@
   boolean shouldAutoClose();
 
   /** True if the TransportProvider needs an executor. */
+  @Deprecated
```

Comment:
Since it is the top interface of this `needsExecutor` thing, please add java doc explaining the reason of deprecation and what to do if you depend on it already.